### DISCUSSION TITLE
Inputs and badges

### DISCUSSION THREAD

**Discussion by jkiga** [twilio-labs/paste#2505] [General] [answered]
Hi again!

This is a little related to my [other discussion](https://github.com/twilio-labs/paste/discussions/2498) I posted, but thought this could be one on its own. 

I understand there isn't a pattern for inputs to have a badges inside and it would be difficult code it there. I have been talking to my engineering team and we were wondering if it would be within best practices to have the badge _on top_ of the input.

For my scenario, the badge is tightly coupled with the information displayed in the read only input. The badge relates to the Messaging Service we want the customer to choose and based on our testing, the recommended badge highly influenced the user's decision. We are iterating on the designs from testing and incorporating the badge is slightly different now. I did some explorations of where the badge could display outside the input, but I'm afraid it might lose its intended context. 

This is what I was hoping/trying to achieve:
![Screen Shot 2022-06-29 at 3 13 07 PM](https://user-images.githubusercontent.com/99448071/176554663-cbe411c5-90f7-41cf-94f0-2a0b8bbafcd3.png)

Here are some other explorations I am considering (left side is first choice after above):
![image](https://user-images.githubusercontent.com/99448071/176554849-38751bee-c6f2-4917-a97b-b5434cc86ba4.png)

I would love to hear feedback or other approaches! Thank you so much.

**Comment by gloriliale** [comment, marked as answer]
Hi @jkiga! Great question, thanks for starting a new thread on it!

First, wanted to confirm that it just isn't possible to put a Badge inside of Input. We use the native html input element for our Input component and the only thing that it takes as a value is a string. 

I definitely understand what you mean by not wanting the context of "Recommended" to get lost. I'll gather the team's thoughts on your examples (thanks for providing them!) and get back to you real soon.

**Reply by gloriliale** [reply]
Hi @jkiga!

The best option here would be to add "Recommended" as text before or after the message service name in the input value itself. I've provided a few screenshots below of what this could look like. 

The reason we recommend (ha) this is because it not only keeps the context visually, but it connects the "Recommended" designation to the service name in an accessible way. In contrast, it'd be difficult for someone using a screen reader to know that a conditionally rendered Badge outside the input box is even related to the input value. Let me know if this is helpful!

![Frame 1](https://user-images.githubusercontent.com/77117846/176565204-bb7c2d41-c9c9-40d7-bab0-61ff68e6895c.png)
![Frame 2](https://user-images.githubusercontent.com/77117846/176565209-51edb1aa-e494-483c-803f-5a23820d8d6b.png)

**Reply by jkiga** [reply]
Hi @gloriliale!

Thank you so much for replying back and taking the time to give me a recommendation (haha). I did play around with this design and took it to my critpod. 

The feedback there was the read only input for the "Select Existing Messaging Service" option was almost there, but it felt it was portrayed in the wrong component. I know this might be headed in a different direction than original posted discussion, but the look and feel that was suggested might align more with uploading a file. Is there an existing pattern for uploading files that I could extract? 


Thank you for all your help!

**Reply by gloriliale** [reply]
Hi @jkiga! I think this hearkens back to your original design with a button and read-only input next to each other. In that case my suggestion would be (and I think we might have touched on this in office hours) to pull this part out of the radio group and have it be a separate section that is labeled are required, pictured below.
<img width="562" alt="image" src="https://user-images.githubusercontent.com/77117846/177435379-91e3b4ca-9d3c-489e-b26a-0e9f67a125e7.png">

**Reply by jkiga** [reply]
Awesome, thank you so much @gloriliale for replying and confirming. I will explore this option!
